packages/airnode-node/src/evm/fulfillments/index.ts

```diff
@@ -74,14 +73,10 @@ function prepareRequestSubmissions<T>(
  * There is a concept of batched requests, but that doesn't work with transactions. See:
  * https://github.com/ethers-io/ethers.js/issues/892#issuecomment-828897859
  */
-async function submitSponsorRequestsSequentially(
-  state: ProviderState<EVMProviderState>,
-  sponsorAddress: string
-): Promise<GroupedRequests> {
+export async function submit(state: ProviderState<EVMProviderSponsorState>): Promise<GroupedRequests> {
   const { AirnodeRrp } = state.contracts;
-  const requestsBySponsorAddress = grouping.groupRequestsBySponsorAddress(state.requests);
-  const requests = requestsBySponsorAddress[sponsorAddress];
-  const sponsorWallet = wallet.deriveSponsorWallet(state.masterHDNode, sponsorAddress);
+  const requests = state.requests;
+  const sponsorWallet = wallet.deriveSponsorWallet(state.masterHDNode, state.sponsorAddress);
```

Comment:
The comment on line 83 and 99 is incorrect - we are not submitting the requests only preparing,(the comment is outdated which is my fault, sry)